kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -297,6 +300,17 @@
       "How many records to write to an intermediate table before performing a merge flush, if " 
       + "upsert/delete is enabled. Can be set to -1 to disable record count-based flushing.";
 
+  public static final String TIME_PARTITIONING_TYPE_CONFIG = "timePartitioningType";
```

Comment:
Is there a reason why this is included in `BigQuerySinkConfig` while `timestampPartitionFieldName` is included in `BigQuerySinkTaskConfig`?

Reply:
`BigQuerySinkTaskConfig` should be for internal-only properties that tasks use but aren't specified by users. The `timestampPartitionFieldName` property and many others should be moved to the `BigQuerySinkConfig` class so that they're visible with GUIs that rely on the `ConfigDef` object returned by `BigQuerySinkConnector::config`, but that's out of scope for this PR. I chose to add `timePartitioningType` to the correct class but leave existing properties where they are.